### ISSUE TITLE
MaximumWidthRequest property behaves differently in Windows and Android

### ISSUE THREAD

**Issue by jlbilh** [dotnet/maui#7366]
### Description

Inner Grid has a MaximumWidthRequest of 600, and is placed inside an outer Grid with a blue background.

On Windows it looks as desired:
![image](https://user-images.githubusercontent.com/102621630/169547810-f971fb07-4e1b-40b4-bd52-81c2c52e9592.png)

Shrink the window and it's still fine:
![image](https://user-images.githubusercontent.com/102621630/169548287-a7ffbd8d-2d10-491f-8ec8-e3ba5df37bcd.png)

On Android it is wrong:
![image](https://user-images.githubusercontent.com/102621630/169548475-78f57881-61ad-4dfe-a77e-a76fe82a26c5.png)

### Steps to Reproduce

New project.

This goes inside the ContentPage:
```
    <Grid BackgroundColor="Blue">
        <Grid MaximumWidthRequest="600" BackgroundColor="White">
            <ScrollView>
                <VerticalStackLayout 
                    Spacing="25" 
                    Padding="30,0" 
                    VerticalOptions="Center">

                    <Image
                        Source="dotnet_bot.png"
                        SemanticProperties.Description="Cute dot net bot waving hi to you!"
                        HeightRequest="200"
                        HorizontalOptions="Center" />

                    <Label 
                        Text="You will rejoice to hear that no disaster has accompanied the commencement of an enterprise which you have regarded with such evil forebodings. I arrived here yesterday; and my first task is to assure my dear sister of my welfare, and increasing confidence in the success of my undertaking."
                        SemanticProperties.HeadingLevel="Level1"
                        FontSize="32"
                        HorizontalOptions="Center" />

                    <Label 
                        Text="Welcome to .NET Multi-platform App UI"
                        SemanticProperties.HeadingLevel="Level2"
                        SemanticProperties.Description="Welcome to dot net Multi platform App U I"
                        FontSize="18"
                        HorizontalOptions="Center" />

                    <Button 
                        x:Name="CounterBtn"
                        Text="Click me"
                        SemanticProperties.Hint="Counts the number of times you click"
                        Clicked="OnCounterClicked"
                        HorizontalOptions="Center" />

                </VerticalStackLayout>
            </ScrollView>
        </Grid>
    </Grid>
```

### Version with bug

Release Candidate 3 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 12

### Did you find any workaround?

Bind MaximumWidthRequest to a view model readonly property that returns Min(600, screenVirtualWidth).

**Comment by kristinx0211** [comment]
can be reproduced on android. repro project:
[MauiApp34.zip](https://github.com/dotnet/maui/files/8751763/MauiApp34.zip)

**Comment by blackzafiqz** [comment]
Can confirmed this happens in scrollview only![Screenshot_2022-08-30-14-25-04-592_com.blackzafiqz.reachme.jpg](https://user-images.githubusercontent.com/17066308/187365241-2fe2f4a8-934e-4b6d-bd27-5d4a425e437f.jpg)

**Comment by nk54** [comment]
I got some problems with MaximumWidthRequest too.

I have a maximum width of 650 on my <Grid> (inside a <ScrollView>) because I don't want my <Entry> to fill all the screen because it looks ugly.
I don't get why it's ok on tablet but on phone, it seems like the maximum width is taken as the requestWidth.

My problem is that on Android Phone, it feels like WidthRequest takes MaximumWidthRequest.
My content goes out of the screen.

See the screenshot below

![image](https://user-images.githubusercontent.com/13755220/201894057-5a826978-66f7-44c7-ae4f-77bc3e96fda6.png)

If I remove the maximum width, here is how it looks (and how it should looks with MaximWidthRequest)
![image](https://user-images.githubusercontent.com/13755220/201892434-92acb1d3-fc21-4de8-b4a0-2889c8b2b425.png)

The problem still exist on the latest .NET 7 released the 8th november 2022

**Comment by Sebastian1989101** [comment]
> In my application's case, the MaximumWidthRequest property works as desired on Android, but on iOS and Mac Catalyst it is ignored completely. E.g. for an item taking the full screen width, setting `MaximumWidthRequest=100` limits the width to 100 only on Android, on Mac and iOS nothing happens when setting that property; I can only limit it for iOS / Mac with `WidthRequest=100`.

I can confirm this... It's so annoying to design your app around MAUI bugs instead around your desired goal...

**Comment by omghb** [comment]
MaximumWidthRequest doesn't work on Android and iOS. 
- Used version: 7.0.81 Service Release 4
- Just put this inside the ContentPage:

```XML
  <Grid BackgroundColor="Blue">
        <Grid MaximumWidthRequest="300" BackgroundColor="LightBlue">
            <Label x:Name="label" Text="{Binding Width, Source={Reference label}}"/>
        </Grid>
    </Grid>
```

On Windows it works as expected. But on Android and iOS the `MaximumWidthRequest="300"` setting is ignored:

![MaximumWidthRequestIssue](https://user-images.githubusercontent.com/47421673/233351662-35a1a825-5387-4722-b227-c719ef6f9db9.png)

**Comment by MV10** [comment]
For a Button, it works on Windows but not Android...

**Comment by samhouts** [comment]
Probably the same situation as https://github.com/dotnet/maui/issues/16464

**Comment by jknaudt21** [comment]
I believe this issue was fixed by #16215. I couldn't reproduce using the sample code using the latest in main.

![image](https://github.com/dotnet/maui/assets/32918747/cc88f338-6a7f-4c47-a898-0471dd82e7d4)


#16464 is another issue and still repros.
